account_invoice_blocking_payment/models/account_invoice.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    blocked = fields.Boolean(store=True)
+
+    draft_blocked = fields.Boolean(store=True)
+
+    @api.multi
+    def create_account_payment_line(self):
+        """
+        A blocked invoice cannot be paid
+        """
+        for rec in self:
+            if rec.draft_blocked or rec.blocked:
+                raise UserError(_(
+                    "A blocked invoice cannot be paid."))
+            super(AccountInvoice, self).create_account_payment_line()
```

Comment:
@ThomasBinsfeld Super should not be called there but outside of the loop and must be returned (inheritance break - e.g.: account_payment_order returns action)